python/cudf/cudf/core/dataframe.py

```diff
@@ -5403,8 +5406,15 @@ def _prepare_for_rowwise_op(self, method, skipna):
             )
             raise ValueError(msg)
 
-        filtered = self.select_dtypes(include=[np.number, np.bool])
-        common_dtype = np.find_common_type(filtered.dtypes, [])
+        is_pure_dt = all(is_datetime_dtype(dt) for dt in self.dtypes)
+
+        if not is_pure_dt:
+            filtered = self.select_dtypes(include=[np.number, np.bool])
+        else:
+            filtered = self.copy(deep=False)
+
+        common_dtype = find_common_type(filtered.dtypes)
```

Comment:
wrapper looks fine - one question, won't you still get `object` here if you have mixed `datetime64` and `int64` columns?

Reply:
Pandas row-wise operation will seem to filter out `datetime` columns if there is a mix between datetime and numeric types:
```python
>>> pdf = pd.DataFrame({"a": pd.Series(["2020-05-01 06:00:00"], dtype="<M8[ms]"), 
                    "b": pd.Series([123456], dtype="int64"),
                   "c":  pd.Series([654321.0], dtype="float64")})
>>> pdf.min(axis=1), pdf.max(axis=1)
(0    123456.0
 dtype: float64, 0    654321.0
 dtype: float64)
```